crates/uv/tests/pip_compile.rs

```diff
@@ -6264,3 +6289,122 @@ fn dynamic_dependencies() -> Result<()> {
 
     Ok(())
 }
+
+/// Allow pre-releases for dependencies of source path requirements.
+#[test]
+fn pre_release_path_requirement() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    // Create an a package that requires a pre-release version of `flask`.
+    let source_tree = TempDir::new()?;
```

Comment:
Can you use a child of the context temporary directory instead? per #2678 

You won't need custom filters then.